epm/Distribution/Client/IndexUtils.hs

```diff
@@ -450,7 +451,7 @@ packageIndexFromCache mkPkg hnd entrs mode = accum mempty [] entrs
       -- from the index tarball if it turns out that we need it.
       -- Most of the time we only need the package id.
       ~(pkg, pkgtxt) <- unsafeInterleaveIO $ do
-        mPatch <- patchedPackageCabalFile pkgid
+        mPatch <- patchedPackageCabalFile pkgid defaultPatchesDir
```

Comment:
The tricky part about this is that sometimes, you *need* the patched Cabal file because the `Build Type` will be set to `Custom` or `Configure` in the original Cabal file that will fail even before it extracts the source code. 

I encountered this behaviour when porting the `time` package and realised only then that I had apply the cabal file patch at the root - in this location. So this will work for almost all packages but for those few packages that have custom build types, we'll have to find some other way of testing them. 

An alternative to testing is to copy over the most recent `eta-hackage` to the into `~/.cabalvm/patches` during the Travis build.